qemu/tests/cpu_model.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu reboot test:
+    1) Start qemu to get cpu model supported by host
+    3) Boot guest with the cpu model
+    4) Check cpu model name in guest
+    5) Check cpu flags in guest(only for linux guest)
+    6) Reboot guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Start qemu to get support cpu model", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    out = vm.monitor.info("cpu-definitions")
+    vm.destroy()
+    model = params["model"]
+    model_pattern = params["model_pattern"]
+    flags = params["flags"]
+    if utils_misc.get_cpu_vendor(verbose=False) == 'GenuineIntel':
+        model_ib = "%s-IBRS" % model
+        flag_ib = " ibpb ibrs"
+        name_ib = ", IBRS( update)?"
+    else:
+        model_ib = "%s-IBPB" % model
+        flag_ib = " ibpb"
+        name_ib = "(with IBPB)"
+
+    models = [x["name"] for x in out if not x["unavailable-features"]]
+    if model_ib in models:
+        cpu_model = model_ib
+        guest_model = model_pattern % name_ib
+        flags += flag_ib
+    elif model in models:
```

Comment:
When host support both model  and model_ib, we test model_id only ?   Is it the expect test strategy ?

Reply:
Yes, they are almost same except 1 or 2 extra flags for model_ib.  It's recommended to use model_ib whenever it's possible.